pgmpy/models/ClusterGraph.py

```diff
@@ -210,6 +209,29 @@ def remove_factors(self, *factors):
         for factor in factors:
             self.factors.remove(factor)
 
+    def get_cardinality(self):
+        """
+        Returns a dictionary with the given factors as keys and their respective
+        cardinality as values.
+        Examples
+        --------
+        >>> from pgmpy.models import ClusterGraph
+        >>> from pgmpy.factors import Factor
+        >>> student = ClusterGraph()
+        >>> factor = Factor(['Alice', 'Bob'], cardinality=[2, 2],
+        ...                 values=np.random.rand(4))
+        >>> student.add_node(('Alice', 'Bob'))
+        >>> student.add_factors(factor)
+        >>> student.get_cardinality()
+        defaultdict(<class 'int'>, {'Bob': 2, 'Alice': 2})
+        
+        """
+        cardinalities = defaultdict(int)
```

Comment:
@yashu-seth We should also have a check somewhere which checks if we have cardinality information for all the variables in the network. It is possible that CPD/factor for all the variables have not yet been added to the network.